composer.json

```diff
@@ -11,16 +11,21 @@
         }
     ],
     "require": {
-        "php": "^5.4 || ^7.0",
-        "php-http/httplug": "^1.1",
+        "php": "^7.0",
+        "php-http/httplug": "^2.0",
         "php-http/message-factory": "^1.0",
         "php-http/message": "^1.6",
         "symfony/options-resolver": "^2.6 || ^3.0 || ^4.0"
     },
     "require-dev": {
-        "phpspec/phpspec": "^2.5 || ^3.4 || ^4.2",
+        "phpspec/phpspec": "^3.4 || ^4.2",
         "guzzlehttp/psr7": "^1.4"
     },
+    "conflict": {
```

Comment:
Those are lower bounds to avoid issues in CI builds. Should not affect `--prefer-lowest`.

Reply:
--prefer-lowest will respect the conflict section. as will other packages, as discussed below. do we use the doctrine/instantiator or sebastian/comparator in our actual code, or only for tests? all things thests should be solved in require-dev rather than here, or we prevent installing this on some systems that are fine with those versions.

Reply:
Both packages are needed by the test suite through PHPSpec/Prophecy; the dev deps tree should not have packages in common with the non-dev deps, so it should be fine. I preferred to use conflicts instead of requiring them explicitly because they are both indirect dependencies that could go away at any time.

Reply:
the problem is that conflicts are also respected by applications using this package. meaning that if my project uses an older version of one of the libraries we mark as conflicting here will not be able to install this package because of the conflict.

conflict should only be used to mark versions of packages that may not be run together with this package. they should not be used to control what is installed for development. if this is all only about --prefer-lowest, i think we either need them in require-dev, or if you want to be more explicit, put them in the DEPENDENCIES of .travis.yml only for the lowest version build - that would be the most explicit way of doing it, imho.